src/components/ui/ClusterEventsToolbar.tsx

```diff
@@ -90,21 +90,42 @@ const ClusterEventsToolbar: React.FC<ClustersListToolbarProps> = ({
     setFilters(getInitialClusterEventsFilters(cluster));
   };
 
-  const onSelect = (type: 'hosts' | 'severity', isChecked: boolean, value: Host['id']) => {
-    const allOtherSelectedHosts =
-      type === 'hosts' ? filters[type].filter((v: string) => v !== value) : [];
+  const onSelect = (
+    type: 'hosts' | 'severity',
+    isChecked: boolean,
+    value: Host['id'] | Event['severity'],
+  ) => {
+    const setNextSelectedValues = (
+      type: 'hosts' | 'severity',
+      isChecked: boolean,
+      value: Host['id'] | Event['severity'],
+      filters: ClusterEventsFiltersType,
+    ) => ({
+      ...filters,
+      [type]: isChecked
+        ? [...filters[type], value]
+        : filters[type].filter((val: string) => val !== value),
+    });
 
-    setFilters({
+    const setNextSelectAllValue = (
+      totalHostsInCluster: number,
+      filters: ClusterEventsFiltersType,
+    ) => ({
       ...filters,
-      [type]: isChecked ? [...filters[type], value] : [...allOtherSelectedHosts],
       selectAll:
         type === 'severity'
           ? filters.selectAll
-          : isChecked &&
-            allOtherSelectedHosts.length === filters.hosts?.length &&
+          : filters.hosts.length === totalHostsInCluster &&
             filters.orphanedHosts &&
             filters.clusterLevel,
     });
+
+    setFilters(
+      _.flow(
```

Comment:
Future bug-fixer will love you ;-)

Reply:
This is just performing plain functional composition (`flow` and `curry` are just there in order to make the order of the  updates more declarative - the functions are applied from left to right inside `flow`, just like a pipe `|` operator would do in `bash`).  Anyway, just to make everybody happy I would like to suggest the following change, which does exactly the same:

```javascript
const updatedFilters = setNextSelectedValues(type, isChecked, value, filters);
const nextFilters = setNextSelectAllValue(cluster.hosts?.length ?? 0, updatedFilters);

setFilters(nextFilters);
```

WDYT?

Reply:
lgtm

Reply:
It's easier to read now

Reply:
resolved in 76b9bf976d7d66d1f83ffb240361ffa48bd45452